add `check_group` support for module files in Tcl syntax

xdelaruelle opened the pull request:
"module-info usergroups" modulefile command has been introduced on Environment Modules version 4.6.0. This commit adds support for check_group in Tcl modulefile when required module tool and version is used.

boegel commented on easybuild/tools/modules.py:
```diff
@@ -1413,6 +1421,7 @@ class Lmod(ModulesTool):
     REQ_VERSION = '6.5.1'
     DEPR_VERSION = '7.0.0'
     REQ_VERSION_DEPENDS_ON = '7.6.1'
+    REQ_VERSION_LUA_CHECK_GROUP = '6.0.8'
```
Does Lmod also support `module-info usergroups`?

Since Tcl syntax can also be used when using Lmod as modules tool, we should also define `REQ_VERSION_TCL_CHECK_GROUP` here?

xdelaruelle replied:
Lmod does not support `module-info usergroups` currently. But if this command is added in the future, it will be easy to update EasyBuild to generate modulefile using this command.

boegel commented:
@xdelaruelle Can you look into fixing the merge conflict here?

xdelaruelle commented:
@boegel Merge conflict has been solved. Everything seem fine on my side. Please let me know if something else should be changed.

xdelaruelle commented:
@boegel, @branfosj I have changed base branch for this PR to 5.0.X and rebased it. Please let me know if something should be changed for this PR to get merged.

boegel reviewed (dismissed):
lgtm

boegel reviewed (approved):
lgtm
